cpmpy/transformations/linearize.py

```diff
@@ -445,6 +437,46 @@ def only_positive_bv(lst_of_expr):
 
     return newlist
 
+def only_positive_bv_sub(expr):
+    """
+        Replaces an pure expression containing NegBoolView with an equivalent expression using only BoolVar.
+        expr is expected to be linearized.
```

Comment:
instead of 'expected to be linearized', be explicit: only accepts variables, sums, wsums; but this does not seem to be true as your bottom block is generic and mentions min/max?